test/clj/cider/nrepl/middleware/info_test.clj

```diff
@@ -163,6 +176,215 @@
+        (is (.startsWith (:doc response) "Returns the URL for a named"))
+        (is (.startsWith (:file response) "jar:file:"))))
+
+    (testing "nested members"
+      (let [response   (session/message {:op "info" :ns (ns-name *ns*) :symbol "toString"})
+            candidates (:candidates response)
+            individual (:java.lang.Exception candidates)]
+        (is (contains? candidates :java.lang.NoSuchFieldException))
+        (is (contains? candidates :java.lang.Package))
+        (is (contains? candidates :java.lang.LinkageError))
+
+        (is (= (:throws individual) []))
+        (is (= (:member individual) "toString"))
+        (is (= (:modifiers individual) "#{:public}"))))
+
+    (testing "boot stuff"
+      (try
+        (System/setProperty "fake.class.path" (System/getProperty "java.class.path"))
+        (let [response (session/message {:op "info" :symbol "as->" :ns "user"})]
+          (is (= (:status response) #{"done"}))
+          (is (= (:ns response) "clojure.core"))
+          (is (= (:name response) "as->"))
+          (is (= (:arglists-str response) "([expr name & forms])"))
+          (is (= (:macro response) "true"))
+          (is (.startsWith (:doc response) "Binds name to expr, evaluates")))        
+        (finally
+          (System/clearProperty "fake.class.path"))))
+
+    ;;TODO: figure out how to actually test this, this seems fake
+    (testing "get protocol info"
+      (let [reply       (session/message {:op "info"
+                                          :ns "cider.nrepl.middleware.info-test"
+                                          :symbol "junk-protocol-client"})
+            status      (:status reply)
+            client-name (:name reply)
+            file        (:file reply)
+            protocol    (:protocol reply)
+            ns          (:ns reply)]
+        (is (= status #{"done"}))
+        (is (= client-name "junk-protocol-client"))
+        (is (.endsWith file "clojure/data.clj"))
+        (is (= protocol "#'clojure.data/Diff"))
+        (is (= ns "cider.nrepl.middleware.info-test")))))
+
+  (testing "eldoc op"
+    (let [response (session/message {:op "eldoc" :symbol "+" :ns "user"})]
+      (is (= (:status response) #{"done"}))
+      (is (= (:eldoc response) [[] ["x"] ["x" "y"] ["x" "y" "&" "more"]])))
+
+    (let [response (session/message {:op "eldoc" :symbol "try" :ns "user"})]
+      (is (= (:status response) #{"done"}))
+      (is (= (:eldoc response) [["try" "expr*" "catch-clause*" "finally-clause?"]])))
+
+    (let [response (session/message {:op "eldoc" :symbol "." :ns "user"})]
+      (is (= (:status response) #{"done"})))
+
+    ;;TODO: Since this is an eldoc java op, shouldn't the args should be :class and :member?
+    (let [response (session/message {:op "eldoc" :symbol "toString" :ns (ns-name *ns*)})]
+      (is (set/subset? #{[] ["this"] ["int"] ["this" "char"]} (set (:eldoc response)))))))
+
+(deftest misses
```

Comment:
Good point. Made the change and pulled out a seperate `missing-eldoc` group of tests.